phpbb/phpbb
# [ticket/16851] Add Amazonbot, AhrefsBot and SemrushBot

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16851
______________
The three bot lists have further been aligned the following way:
- Complement each list with all bots present in another list, in case. There might be a reason why a bot has been removed from a list, but then it should be removed from all list, for consistency.
- Remove a bots version integer from user agent match string, if present in one list but missing in another. Generally, since versions change, it makes sense to match it, so the entry has a chance to remain valid over a longer time.
- Add trailing slash to user agent match string, if missing in one list but present in another. Nearly all bots have a slash in their user agent name, separating the bots name from the version string. Including it into the match makes sense to lower the chance of false positives a little.
- Remove `+(` from Baidu spider, which is nowadays not part of its user agent. Instead, there are many different Baidu spiders, with or without a dash + content type prefixed. The used string matches all of them: https://qpsoftware.net/blog/baiduspider
- Add the wildcard to all W3C(*)_Validator strings, although it doesn't seem to be required nowadays: http://validator.w3.org/services

More information about the added/changed bots:
- Amazonbot: https://developer.amazon.com/de/support/amazonbot
- AhrefsBot: https://ahrefs.com/robot
- SemrushBot: https://www.semrush.com/bot/
- Baiduspider: https://qpsoftware.net/blog/baiduspider
- Exabot: https://www.webrobots.de/exabot/
- W3C_Validator: http://validator.w3.org/services
